### DISCUSSION TITLE
Multiple subdomains all using default.crt

### DISCUSSION THREAD

**Discussion by colfin22** [nginx-proxy/nginx-proxy#2169] [Q&A] [answered]
Apologies if this is in the wrong place I am using nginx-proxy with acme-companion but it is not working with SSL on port 443 for any domain when I had it working before with jwilder/docker-letsencrypt-nginx-proxy-companion before I completely removed nginx as part of a rebuild. Acme-companion is successfully getting the certs for all subdomains (after as few days hitting the 5 limit) but I cannot get SSL on port 443 to work. When I check the nginx default.conf (snippet below) I see that all are set to use the same default.crt and default.key when I have ,crt and .key for each subdomain. What am I doing wrong

     
[docker-compose.yml.txt](https://github.com/nginx-proxy/nginx-proxy/files/10716962/docker-compose.yml.txt)
[default.conf.txt](https://github.com/nginx-proxy/nginx-proxy/files/10716963/default.conf.txt)

**Comment by rhansen** [comment]
Double check your volumes.  I recommend against using `volumes_from` because its semantics are unclear and it is not supported in compose file v3.  Use `docker exec $nginx_proxy_container_name ls -l /etc/nginx/certs` to verify that your nginx-proxy container can see the certs generated by the acme-companion container.

Also, make sure the certificate filenames match what nginx-proxy expects.  You have `CERT_NAME` set to `embycert`, so the files should be named `/etc/nginx/certs/embycert.crt` and `/etc/nginx/certs/embycert.key`.  (See https://github.com/nginx-proxy/acme-companion/issues/997.)

**Comment by colfin22** [comment]
Just confirmed and Nginx is seeing all the certs

**Reply by rhansen** [reply]
Do the filenames match your `CERT_NAME` environment variable?  `/etc/nginx/certs/embycert.crt` and `/etc/nginx/certs/embycert.key`

**Reply by colfin22** [reply]
No, they are `LETSENCRYPT_HOST` or `VIRTUAL_HOST` not sure which as they are both the same. It also did not work before I added `CERT_NAME`

**Reply by rhansen** [reply]
> No, they are `LETSENCRYPT_HOST` or `VIRTUAL_HOST` not sure which as they are both the same.

OK, that's why it's not working.  Change `CERT_NAME` to match or leave `CERT_NAME` unset (the default should work).

> It also did not work before I added `CERT_NAME`

Did not work how?

**Comment by colfin22** [comment, marked as answer]
Okay, I have it working, commenting out the CERT_NAME fixed the issue. Thank you for your time
